src/libxtp/moldenwriter.cc

```diff
@@ -0,0 +1,104 @@
+#include "votca/xtp/moldenwriter.h"
+
+#include "votca/xtp/basisset.h"
+
+namespace votca {
+namespace xtp {
+
+void MoldenWriter::writeAtoms(const Orbitals& orbitals,
+                              std::ofstream& outFile) const {
+  for (auto& atom : orbitals.QMAtoms()) {
+    Eigen::Vector3d pos = atom.getPos();
+    outFile << boost::format("%-4s %5d %5d %22.12e %22.10e %22.10e\n") %
+                   atom.getElement() % (atom.getId() + 1) %
+                   atom.getTotalNuccharge() % pos[0] % pos[1] % pos[2];
+  }
+}
+
+void MoldenWriter::writeMOs(const Orbitals& orbitals,
+                            std::ofstream& outFile) const {
+
+  Eigen::VectorXd energies = orbitals.MOs().eigenvalues();
+
+  OrbReorder reorder(_reorderList, _multipliers);
+
+  Eigen::MatrixXd moCoefficients = orbitals.MOs().eigenvectors();
+
+  reorder.reorderOrbitals(moCoefficients, _basis);
+
+  for (Index i = 0; i < orbitals.getBasisSetSize(); i++) {  // over columns
+    outFile << "Sym= \n";
+    outFile << boost::format("Ene= %-20.12e\n") % energies[i];
+    outFile << "Spin= Alpha\n";
+    outFile << boost::format("Occup= %-5.2f\n") %
+                   (2 * (i < orbitals.getLumo()));
+    for (Index j = 0; j < orbitals.getBasisSetSize(); j++) {
+      outFile << boost::format("%5d %22.12e\n") % (j + 1) %
+                     moCoefficients(j, i);
+    }
+  }
+}
+
+void MoldenWriter::writeBasisSet(const Orbitals& orbitals,
+                                 std::ofstream& outFile) const {
+  if (orbitals.hasDFTbasisName()) {
+
+    for (auto& atom : orbitals.QMAtoms()) {
```

Comment:
```suggestion
    for (const auto& atom : orbitals.QMAtoms()) {
```

Reply:
why do you not simply write out the aobasis instead of the basisset?

Reply:
the aobasis is the basisset applied to the qmatoms

Reply:
Access to gaussian primitives is easier in the basisset and the contractions are not yet normalised in the basisset, hence you can check that the right decay contraction pair is written to the molden file.